### ISSUE TITLE
Distribute on F-Droid?

### ISSUE THREAD

**Issue by paulcmal** [Jigsaw-Code/intra#9]
Hi! Like many people i'm not using the Play Store (or any Google services) on my Android phone, but I'm very interested in trying out DNS over HTTPS on my phone.

I would like to compile the project, but i think it would be nice to distribute the application outside of Google. [F-Droid](https://f-droid.org) is a free-software app manager for Android. The main repo at f-droid.org contains only free software, and there may be a few steps to meet the [inclusion policy](https://f-droid.org/en/docs/Inclusion_Policy/).

Is this something you would be willing to do ? Thanks for building intra :)

**Comment by paidfocusgroups** [comment]
I'm sure the "F" in F-Droid means free.....

😎

**Comment by mtimvik** [comment]
It would be Great if I can install this app from F-droid.

**Comment by TPS** [comment]
[I've RFP'ed @ F-Droid](https://gitlab.com/fdroid/rfp/issues/735), but this usually takes several months to publish (even if build is perfectly suited), due to lack of volunteers & hardware.

**Comment by TPS** [comment]
@Jigsaw-Code So far, there are a [# of initial concerns w/ this, mostly due to proprietary (mostly Google) components](https://gitlab.com/fdroid/rfp/issues/735#note_106719226).

**Comment by cjhenck** [comment]
Hi all - thanks for the feedback.

If someone is able to create a PR that would enable us to easily create an F-Droid build with each release while maintaining existing functionality for the Play Store then we'd be happy to consider merging it.

**Comment by TPS** [comment]
So, the current source is nixed due to [the 🤖/scanner's concerns](https://gitlab.com/fdroid/rfp/issues/735#note_106934133), so, as @cjhenck [mentioned](https://github.com/Jigsaw-Code/Intra/issues/9#issuecomment-427512343), work is required to comply w/ F-Droid.

**Comment by osdroid** [comment]
Ohh by looking at the scanner concerns, as far as I can see it, the only problem is the use of firebase? (which also requires gms)? And the google decision not to use external repositories? If those are the problems, I think it can be worked out quite easily...

**Comment by paulcmal** [comment]
> Intra uses Firebase so that we can get notified if the app is performing poorly or crashing.

So basically disabling telemetry would produce a self-contained free-software build that would meet F-Droid's requirements, or am i missing something?

**Comment by cjhenck** [comment]
I'm not very familiar with F-Droid's requirements, but that seems to be their primary concern.

**Comment by Ilia1** [comment]
hmm, if you use the [gun](http://gun.js.org/), the policy f-droid will not be violated? or another from [this list](https://alternativeto.net/software/firebase/?license=opensource)

**Comment by cjhenck** [comment]
We don't use firebase for the database, but as before we are open to someone creating a firebase-free branch or build. It should be fairly straightforward.

**Comment by WPFilmmaker** [comment]
Are there any updates on this? The ticket on gitlab pointed out by TPS is closed and f-droid has no intra app :/

**Comment by TPS** [comment]
The most current release on GPlay has [the same concerns](https://reports.exodus-privacy.eu.org/reports/88556), so F-Droid has no reason to reconsider.

**Comment by cjhenck** [comment]
I want to reiterate that we're willing to accept patches to make an F-Droid compatible build.  This should be as simple as adding an intermediate class for FirebaseAnalytics and having it perform no operation in a new [build flavor](https://developer.android.com/studio/build/build-varants)

**Comment by linsui** [comment]
I open an MR in F-Droid side. https://gitlab.com/fdroid/fdroiddata/-/merge_requests/11000 It works but I thought it would be cleaner to add a flavor here. I'd like to also add fastlane here.

> I want to reiterate that we're willing to accept patches to make an F-Droid compatible build. This should be as simple as adding an intermediate class for FirebaseAnalytics and having it perform no operation in a new [build flavor](https://developer.android.com/studio/build/build-varants)

@cjhenck After 3 years, is that still vaild? I thougth I need to sign a CLA? Can I send a patch to you instead?

**Comment by cjhenck** [comment]
> > I want to reiterate that we're willing to accept patches to make an F-Droid compatible build. This should be as simple as adding an intermediate class for FirebaseAnalytics and having it perform no operation in a new [build flavor](https://developer.android.com/studio/build/build-varants)
> 
> @cjhenck After 3 years, is that still vaild? 

Yes, if it were controlled by a flag. (We wouldn't want to remove firebase for the main build, so the changes would need to be enabled/disabled.)

> I thougth I need to sign a CLA? Can I send a patch to you instead?

Yes, you need to sign a CLA for us to accept it. May I ask the concern with the CLA? Is it that it requires a Google account?

If you made a fork with an appropriate license (such as the existing one  - Apache 2) then maybe we could incorporate the changes.

**Comment by linsui** [comment]
> Yes, you need to sign a CLA for us to accept it. May I ask the concern with the CLA? Is it that it requires a Google account?

I don't have that. I did have one but now it requires my phone number so I can't access it anymore.

> If you made a fork with an appropriate license (the existing license is Apache 2) then maybe we could incorporate the changes. (Assuming it's not too much work.)

Sounds good to me. Thanks!

**Comment by linsui** [comment]
@cjhenck https://github.com/linsui/Intra/tree/fdroid I create a fork with fdroid flavor and fastlane structure.

**Comment by rugk** [comment]
welll… then maybe first (try to) contribute that back upstream? Otherwise, you may open issues on your fork, maintain it and then publish it on F-Droid…

**Comment by linsui** [comment]
@rugk The CLA is the problem.

> Yes, you need to sign a CLA for us to accept it. May I ask the concern with the CLA? Is it that it requires a Google account?

> If you made a fork with an appropriate license (the existing license is Apache 2) then maybe we could incorporate the changes. (Assuming it's not too much work.)

**Comment by ignoramous** [comment]
@linsui  where are `outline-go-tun2socks` being pulled from for Intra (particularly, how is it built?)? The package size is 12MB (when I expected it to be higher because gomobile should be building an `outline-go-tun2socks` lib per arch target (x86, x64, arm64, and arm32) and that can't be just 12MB?

**Comment by linsui** [comment]
See https://gitlab.com/fdroid/fdroiddata/-/commit/1e2cb715a2acc6132eb40ae96c42a68c7d684e15#f9241e17bfd1ddc18e18fd31d7b986cae8d85812_0_64 The [tun2soucks aar in this repo](https://github.com/Jigsaw-Code/Intra/blob/master/Android/tun2socks/tun2socks.aar) is 10MB. I see no problem. The gomobile files are there. You can unzip the apk and check them.

```
-rw-rw-rw-  0.0 unx  5437048 b- defN 81-Jan-01 01:01 lib/arm64-v8a/libgojni.so
-rw-rw-rw-  0.0 unx  5237076 b- defN 81-Jan-01 01:01 lib/armeabi-v7a/libgojni.so
-rw-rw-rw-  0.0 unx  5303120 b- defN 81-Jan-01 01:01 lib/x86/libgojni.so
-rw-rw-rw-  0.0 unx  5722936 b- defN 81-Jan-01 01:01 lib/x86_64/libgojni.so
```

**Comment by cjhenck** [comment]
Hi @linsui - I'm slightly concerned about needing to update near-duplicate versions of similar files. I was hoping there might be some way to use mocks, though I can't find any off-the-shelf mocks for Firebase analytics other than for javascript/typescript. Apologies for the late engagement.

**Comment by linsui** [comment]
@cjhenck Some apps use their own mock but I don't know if there if any standalong implementation. How about this method: https://github.com/zhanghai/MaterialFiles/blob/f13a397ad5f86b8e60cbdf14c2f510f887b7a411/app/src/main/java/me/zhanghai/android/files/provider/archive/archiver/ArchiveException.kt#L8?

**Comment by Korb** [comment]
[Done](https://f-droid.org/packages/app.intra):

![Image](https://github.com/user-attachments/assets/b61b693d-5959-4016-9a17-1e5315cd917b)

**Comment by licaon-kter** [comment]
this can be closed?
